channel.rb

```diff
@@ -45,22 +45,38 @@ def retrieve_packages
     packages = {}
     puts "Fetching packages for channel https://#{@domain}/#{@channel_name}..."
     channeldata = HTTParty.get("https://#{@domain}/#{@channel_name}/channeldata.json")["packages"]
+
     benchmark = Benchmark.measure do
       ARCHES.each do |arch|
-        blob = HTTParty.get("https://#{@domain}/#{@channel_name}/#{arch}/repodata.json")["packages"]
-        blob.each_key do |key|
-          version = blob[key]
-          package_name = version["name"]
-
-          unless packages.key?(package_name)
-            package_data = channeldata[package_name]
-            packages[package_name] = base_package(package_data, package_name)
-          end
+        repo_data_url = "https://#{@domain}/#{@channel_name}/#{arch}/repodata.json"
+        response = HTTParty.get(repo_data_url)
+
+        parsed_response = JSON.parse(response.body)
+
+        blobs = [parsed_response["packages"], parsed_response["packages.conda"]]
+
+        blobs.each do |blob|
+          next if blob.nil?
 
-          packages[package_name][:versions] << release_version(key, version)
+          blob.each_key do |key|
+            version = blob[key]
+            package_name = version["name"]
+
+            if channeldata[package_name].nil?
+              puts "Unable to get channel data for package #{package_name} from url: #{repo_data_url}"
+              next
+            end
+
+            unless packages.key?(package_name)
+              package_data = channeldata[package_name]
+              packages[package_name] = base_package(package_data, package_name)
+            end
+
+            packages[package_name][:versions] << release_version(key, version)
```

Comment:
and a followup to the prev suggestion: 

```suggestion
            packages[package_name][:versions] << release_version(artifact_filename, version)
```